### PR TITLE
i refactored uses of [db] and [table]

### LINKED ISSUE #17727: `DatabaseName` and `TableName` should be used for `$db` and `$table`
The `PhpMyAdmin\Dbal\DatabaseName` and `PhpMyAdmin\Dbal\TableName` classes should be used as a type for the database and table names instead of using plain strings.

```diff
- public function foo(string $db, string $table)
+ public function foo(DatabaseName $db, TableName $table)
```

### PR DESCRIPTION
**Pull request by Adams-Ijachi**
### Description

Please describe your pull request.
i refactored some instances $Global[db] and $Global[table] to use a value object

Related to #17727

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Comment by Adams-Ijachi** [comment]
I just made this commit , i plan on making more but wanted to know if i am on the right path

File: libraries/classes/Controllers/CheckRelationsController.php [hunk 1]
```diff
@@ -43,7 +44,7 @@ public function __invoke(ServerRequest $request): void
 
         // If request for creating all PMA tables.
         if (isset($fixAllPmaDb)) {
-            $this->relation->fixPmaTables($GLOBALS['db']);
+            $this->relation->fixPmaTables(DatabaseName::fromValue($GLOBALS['db']));
```
Comment (MauricioFauth) [inline libraries/classes/Controllers/CheckRelationsController.php:47]: This method is expecting a string, not a DatabaseName object. You should first create the object, then pass the name as string. For example:

```suggestion
            $db = DatabaseName::tryFromValue($GLOBALS['db']);
            $this->relation->fixPmaTables($db !== null ? $db->getName() : '');
```

File: libraries/classes/Controllers/CheckRelationsController.php [hunk 2]
```diff
@@ -56,7 +57,7 @@ public function __invoke(ServerRequest $request): void
         $relationParameters = $this->relation->getRelationParameters();
 
         $this->render('relation/check_relations', [
-            'db' => $GLOBALS['db'],
+            'db' => DatabaseName::fromValue($GLOBALS['db']),
```
Comment (MauricioFauth) [inline libraries/classes/Controllers/CheckRelationsController.php:60]: The key `'db'` is expecting a string value here.

File: libraries/classes/Controllers/HomeController.php [hunk 3]
```diff
@@ -224,8 +226,8 @@ public function __invoke(ServerRequest $request): void
         $git = new Git($this->config->get('ShowGitRevision') ?? true);
 
         $this->render('home/index', [
-            'db' => $GLOBALS['db'],
-            'table' => $GLOBALS['table'],
+            'db' => DatabaseName::fromValue($GLOBALS['db']),
+            'table' => TableName::fromValue($GLOBALS['table']),
```
Comment (MauricioFauth) [inline libraries/classes/Controllers/HomeController.php:230]: The `'db'` and `'table'` keys are expecting a string value here.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

**Comment by MauricioFauth** [comment]
Hi @Adams-Ijachi. Thank you for your contribution.
This pull request was replaced by https://github.com/phpmyadmin/phpmyadmin/pull/17829.
I hope to see more contributions from you.

**Pull request closed without merging**